### PR TITLE
[back/front] add form intake draft default values (#14057)

### LINKED ISSUE #14057: Workflow: Ability to define default value for drafts created through Form Intake
## Use case

When starting a workflow with form intake, it is important, in the form intake definition, to be able to define some default values, that would override draft default. This way, we can be sure that the RBAC when entering in a workflow is correct. 

## Proposed Solution

- Define Author of the draft: author must be either an instance selected or dynamically retrieved by defining the same org than the org author of the main entity
- Define Status in which the draft should be created at
- Define the authorized members applied when the draft is created
   - Authorize members must support the intersection between org and group to be able to express something like (analyst of Org A can manage)
   - Org should be either selected manually or define dynamically, by providing the same option than in playbooks Org (author)
  
## If the feature request is approved, would you be willing to submit a PR?

Yes / No (Help can be provided if you need assistance submitting a PR)

### PR DESCRIPTION
**Pull request by OctaveLaventure**
### Proposed changes

Allow users to provide default values for draft creation via form intakes.

This PR adds fields in the form intake creation form, and in the form intake shown to the end user.
It updates the end-to-end logic between and after these forms so form intakes default values are applied to the resulting draft entity.
Multiple little edge cases have been handled in separate commits, with many regression tests.

A user can now create a form intake to allow other users to create drafts with pre-defined:
- Author (A valid author Identity or dynamically select the form intake's main entity's author)
- Authorized Members (with the possibility to use the user's organization, or an organisation+group intersection which also applies to dynamic organization options)
- Participants
- Assignee 
- Description

These fields can be at the same time pre-defined AND made EITHER read-only OR editable/clearable.

The read-only fields can still be edited by users with BYPASS.

### Related issues

* https://github.com/OpenCTI-Platform/opencti/issues/14057

### Checklist

- [x] I consider the submitted work as finished
- [x] I tested the code for its functionality
- [x] I wrote test cases for the relevant uses case (coverage and e2e)
- [ ] I added/update the relevant documentation (either on github or on notion)
- [x] Where necessary I refactored code to improve the overall quality

### CODE REVIEW

File: opencti-platform/opencti-graphql/src/modules/form/form-domain.ts [hunk 1]
```diff
@@ -389,6 +389,189 @@ export const completeEntity = (entityType: string, entity: StoreEntity) => {
...
+          const currentGroupRestrictions = existing.groups_restriction_ids ?? [];
+          const mergedGroupRestrictions = groupsRestrictionIds
+            ? Array.from(new Set([...currentGroupRestrictions, ...groupsRestrictionIds]))
+            : undefined;
+
+          authorizedMembersMap.set(org.internal_id, {
+            ...existing,
+            access_right: existing.access_right || accessRight,
+            groups_restriction_ids: mergedGroupRestrictions,
+          });
```
[the remark this answers is not in the document]
Comment (OctaveLaventure) [reply opencti-platform/opencti-graphql/src/modules/form/form-domain.ts]: if a rule explicitly has no group restriction, the org gets unrestricted access, this is the wanted behavior. Fixing it that way would potentially cause unwanted over-restriction. I'll add a comment.

File: opencti-platform/opencti-graphql/tests/01-unit/domain/draft-workspace-test.ts [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+    // sanitized members should have empty groups restriction removed
+    await addDraftWorkspace(mockContext, mockUser, input);
+
+    expect(middleware.createEntity).toHaveBeenCalledWith(
+      expect.anything(),
+      expect.anything(),
+      expect.objectContaining({
+        authorized_members: expect.arrayContaining([
+            expect.objectContaining({ id: 'user2', access_right: 'view' }), // groups_restriction_ids implicitely removed
+            expect.objectContaining({ id: 'user3', access_right: 'admin', groups_restriction_ids: ['group1'] }),
```
[the remark this answers is not in the document]
Comment (OctaveLaventure) [reply opencti-platform/opencti-graphql/tests/01-unit/domain/draft-workspace-test.ts]: fixed

File: opencti-platform/opencti-front/src/private/components/data/forms/FormSchemaEditor.tsx [hunk 3]
```diff
@@ -1570,6 +1685,315 @@ const FormSchemaEditor: FunctionComponent<FormSchemaEditorProps> = ({
...
+                      </Select>
+                    </FormControl>
+                    {formData.draftDefaults?.author?.type === 'static' && (
+                      <Box style={{ paddingLeft: 20, paddingBottom: 10 }}>
+                        <CreatedByField
+                          name="authorDefaultIdentity"
+                          label={t_i18n('Default author identity')}
+                          style={{ width: '100%', marginBottom: 20 }}
+                          setFieldValue={() => {}}
+                        />
```
[the remark this answers is not in the document]
Comment (OctaveLaventure) [reply opencti-platform/opencti-front/src/private/components/data/forms/FormSchemaEditor.tsx]: It seemed to work as selecting an identity works thanks to Formik's Field inner mechanism, but no setFieldValue could indeed break the selection when an identity is **created** directly from the input. Fixed it 👍🏻

**Review by delemaf — APPROVED** [review]
Tested with and without feature flag 🚀
